Connectors/Autocad/Speckle.Connectors.AutocadShared/HostApp/AutocadColorManager.cs

```diff
@@ -0,0 +1,126 @@
+    string name = color.ColorNameForDisplay;
+
+    ColorProxy colorProxy = new(argb, id, name) { objects = new() };
+
+    // INFO: this index is an Autocad internal index for set rgb values
+    // https://gohtx.com/acadcolors.php
+    if (color.IsByAci)
+    {
+      colorProxy["autocadColorIndex"] = (int)color.ColorIndex;
+    }
+
+    return colorProxy;
+  }
+
+  public List<ColorProxy> UnpackColors(List<AutocadRootObject> rootObjects, List<LayerTableRecord> layers)
+  {
+    Dictionary<string, ColorProxy> colorProxies = new();
+
+    // Stage 1: unpack colors from objects
+    foreach (AutocadRootObject rootObj in rootObjects)
+    {
+      Entity entity = rootObj.Root;
+
+      // skip any objects that inherit their colors
+      if (!entity.Color.IsByAci && !entity.Color.IsByColor)
+      {
+        continue;
+      }
+
+      // assumes color names are unique
+      string colorId = entity.Color.ColorNameForDisplay;
+
+      if (colorProxies.TryGetValue(colorId, out ColorProxy value))
+      {
+        value.objects.Add(rootObj.ApplicationId);
+      }
+      else
+      {
+        ColorProxy newColor = ConvertColorToColorProxy(entity.Color, colorId);
+        newColor.objects.Add(rootObj.ApplicationId);
+        colorProxies[colorId] = newColor;
+      }
+    }
+
+    // Stage 2: make sure we collect layer colors as well
+    foreach (LayerTableRecord layer in layers)
+    {
+      // assumes color names are unique
+      string colorId = layer.Color.ColorNameForDisplay;
+      string layerId = layer.Handle.ToString();
+
+      if (colorProxies.TryGetValue(colorId, out ColorProxy value))
+      {
+        value.objects.Add(layerId);
+      }
+      else
+      {
+        ColorProxy newColor = ConvertColorToColorProxy(layer.Color, colorId);
+        newColor.objects.Add(layerId);
+        colorProxies[colorId] = newColor;
+      }
+    }
+
+    return colorProxies.Values.ToList();
+  }
+
+  public AutocadColor ConvertColorProxyToColor(ColorProxy colorProxy)
+  {
+    AutocadColor color = colorProxy["autocadColorIndex"] is long index
+      ? AutocadColor.FromColorIndex(ColorMethod.ByAci, (short)index)
+      : AutocadColor.FromColor(System.Drawing.Color.FromArgb(colorProxy.value));
+
+    return color;
+  }
+
+  /// <summary>
+  /// Parse Color Proxies and stores in ObjectColorIdMap the relationship between object ids and colors
+  /// </summary>
+  /// <param name="colorProxies"></param>
+  /// <param name="onOperationProgressed"></param>
+  public void ParseColors(List<ColorProxy> colorProxies, Action<string, double?>? onOperationProgressed)
+  {
+    var count = 0;
+    foreach (ColorProxy colorProxy in colorProxies)
+    {
+      onOperationProgressed?.Invoke("Converting colors", (double)++count / colorProxies.Count);
+      foreach (string objectId in colorProxy.objects)
+      {
+        AutocadColor convertedColor = ConvertColorProxyToColor(colorProxy);
+
+        if (!ObjectColorsIdMap.TryGetValue(objectId, out AutocadColor _))
```

Comment:
If we won't use get value of `TryGetValue` we could use `ContainsKey` too, but it is minor.